### ISSUE TITLE
guest os customization not executed with computer_name parameter

### ISSUE THREAD

**Issue by keeslouwen** [terraform-providers/terraform-provider-vcd#338]
Hello, 

I am testing with the latest provider vcd 2.5.0 release (not released yet). I notice some unexpected behaviour with templates and guest os customization in vcd. 

I have a template in a catalog which contains guest os customization including a script. When I use this template to create a new VM with the new parameter computer_name the guest os customization isn't executed and the script isn't available in the created VM. Only the computer_name / hostname is set correctly.

### Terraform Version
Terraform v0.12.9
+ provider.vcd v2.5.0

### Affected Resource(s)
vcd_vapp_vm

### Terraform Configuration Files
resource "vcd_vapp_vm" "app01" {
  vapp_name     = "${vcd_vapp.v8ucase1.name}"
  name          = "app01"
  computer_name = "app01"
  catalog_name  = "v8demo"
  template_name = "centos7-vcd-template"
  memory        = 2048
  cpus          = 2
  customization {
    force = true
  }
  metadata = {
    role    = "v8app"
    env     = "test"
    version = "v1"
  }
  network {
    type               = "org"
    name               = "v8ucase1"
    ip                 = "10.11.105.162"
    ip_allocation_mode = "MANUAL"
    is_primary         = true
  }
  depends_on = ["vcd_vapp.v8ucase1"]
}

### Expected Behavior
I would expect the guest os customization properties is 'copied' from the template to the new VM and executed. computer name / hostname is set based upon the parameter computer_name, auto generated password is set and script is executed. 

### Actual Behavior
Only the computer name is correctly set, all other guest os customization isn't executed, so no auto generated password and no script available/executed.

if I provide the script via the initscript parameter, the script is available and executed correctly. Only auto generate password is not executed.  

### Steps to Reproduce
terraform apply

### References
This relates to issues #334 and #273. 

Thanks for the provider and your time. Please let me know if you need more info. 
rgds.

**Comment by lvirbalas** [comment]
CC: @vbauzysvmware @Didainius

**Comment by vbauzysvmware** [comment]
Thank you @keeslouwen for testing it. We will fix that.

**Comment by vbauzysvmware** [comment]
Hi @keeslouwen, could be possible for you to verify fix provided in PR? https://github.com/terraform-providers/terraform-provider-vcd/pull/347

**Comment by keeslouwen** [comment]
Hi, yes I will do and let you know.

**Comment by keeslouwen** [comment]
Hello, I have tested with the different settings and it seems to work oke with computer_name set. If I also set the initscript in the terraform config file, the auto generate password function is not executed.

Results with terraform config mentioned earlier and different related settings are: 
test1:
computer_name=xxx  - result: hostname correctly set to computer_name parameter value; initscript from template taken and correctly executed; auto generate password oke; 
test2: 
computer_name=xxx; initscript="same initscript as template"  - result: hostname correctly set to computer_name parameter value; initscript from config file taken and correctly executed; auto generate password NOT oke; 

So for my initial issue and use case the fix is oke;  

thanks!
